build-tools/cache_samples.sh

```diff
@@ -103,7 +103,7 @@ devfileEntriesFile=$1
 samplesDir=$2
 
 for sample in $(yq e '(.samples[].name)' $devfileEntriesFile); do
-  mkdir $samplesDir/$sample
+  mkdir -p $samplesDir/$sample
```

Comment:
```suggestion
  mkdir -p .$samplesDir/$sample
```

When trying to test it locally I was running into the following error:
`mkdir: /samples: Read-only file system`

After doing some digging it seems it may be a MacOS issue, adding the `.` to tell the script it is creating the directories based on the location of the script fixed the issue, I think it is something to do with MacOS and roots. When you exclude the `.` it thinks you are creating from the absolute root, see [this StackOverflow post](https://stackoverflow.com/a/62818209) about it.

Not sure if that messes with the functionality of the script for other OS though.

Reply:
Addressing this here: https://github.com/devfile/registry-support/pull/239/commits/5bb20c06266bb35d446f2370fd53de400d557921